### PR TITLE
Call valid move

### PR DESCRIPTION
**Pull request by koglesby**
- piece update action calls move_to! to determine whether to update attributes and change the position
- move_to! calls valid_move?
- fixed bugs in obstructed? method, moved horizontal_check, vertical_check, and diagonal_check to private
- fixed bug in valid_move? for King, valid_move? for Rook calls did_not_move?
- fixed test for pawn when obstructed

It seems like it should be relatively simple, but I couldn't figure out how to successfully show an alert only when a player attempts an invalid move

### CODE REVIEW

**Comment by koglesby** [comment]
Still a bug in diagonal_check section of obstructed? method
Pieces do not fly back after performing an invalid move, but if you refresh the page, the piece move will stick if it is valid

File: app/views/games/show.html.erb [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+          horizontal_position: x,
+          vertical_position: y
+        } };
+        $(ui.draggable).detach().css({top: 0,left: 0}).appendTo(this);
+        $.ajax({
+          type: 'PUT',
+          url: desiredUrl,
+          data: piece,
+          dataType: 'json'
+        });
```
Comment (dtrypt) [inline app/views/games/show.html.erb]: I still don't know how to fix the fly back issue for invalid moves... somehow the javascript needs to talk to the move_to method in the controller and then only stick to the new spot if it passes validation. I have no idea how to do that.
For now add this so the page reloads after each move:
```
          dataType: 'json'
        }).done(function(){
          window.location.reload();
        });
      }
```

**Comment by koglesby** [comment]
- Fixed the issue in diagonal_check for obstructed? method
- Whole page reloads automatically after every piece move
- Disabled the click-link to pieces show page, so now clicking on a piece without dragging it anywhere does not attempt to take you the page

**Review by dtrypt — APPROVED** [review]
Nice getting it all working correctly and fixing my mistakes in the diagonal check. I tested it out a lot and it works well. Also nice job with with getting rid of the piece show page and making the pieces inactive links. I think this should be merged in and we can close out the capture logic PR as well.

**Comment by rouxcaesar** [comment]
I agree with Russ and the code seems to be running well. Once this has been merged into the master branch, I'll go ahead and close the Capture Logic branch. Great job!

**Review by rouxcaesar — APPROVED** [review]
I agree with Russ and the code seems to be running well. Once this has been merged into the master branch, I'll go ahead and close the Capture Logic branch. Great job!

File: app/controllers/pieces_controller.rb [hunk 2]
```diff
@@ -1,32 +1,16 @@
...
-
   def update
     @piece = Piece.find(params[:id])
-    @piece.update_attributes(piece_params)
-    redirect_to game_path(@piece.game)
+    @piece.move_to!(piece_params)
+    respond_to do |format|
+      format.html { redirect_to game_path(@piece.game) }
+      format.json { render json: @piece }
+    end
```
Comment (sungwon) [inline app/controllers/pieces_controller.rb:8]: Nice.

**Review by sungwon — APPROVED** [review]
Nice! Looks good.

**Pull request merged**